tests/tensorflow/quantization/test_builder_state.py

```diff
@@ -0,0 +1,326 @@
+
+    model, compression_ctrl = create_compressed_model_and_algo_for_test(get_basic_conv_test_model(),
+                                                                        config, force_no_init=True)
+    assert isinstance(compression_ctrl, QuantizationController)
+
+    quantizer_setup = gen_setup_spy.spy_return
+    compression_callbacks = create_compression_callbacks(compression_ctrl, log_tensorboard=False)
+    dataset_len = 8
+
+    dummy_x = tf.random.normal((dataset_len,) + model.input_shape[1:])
+    dummy_y = tf.random.normal((dataset_len,) + model.output_shape[1:])
+
+    model.compile(loss=tf.losses.CategoricalCrossentropy())
+
+    ckpt_path = tmp_path / 'checkpoint'
+    ckpt = tf.train.Checkpoint(model=model, compression_state=TFCompressionState(compression_ctrl))
+    model.fit(dummy_x, dummy_y,
+              epochs=5,
+              batch_size=2,
+              callbacks=[CheckpointManagerCallback(ckpt, str(ckpt_path), save_freq), *compression_callbacks])
+
+    assert sorted(os.listdir(ckpt_path)) == REF_CKPT_DIR[save_freq]
+
+    new_compression_state = load_compression_state(ckpt_path)
+
+    new_model, new_compression_ctrl = create_compressed_model_and_algo_for_test(get_basic_conv_test_model(),
+                                                                                config, new_compression_state)
+    new_model.compile(loss=tf.losses.CategoricalCrossentropy())
+    new_ckpt = tf.train.Checkpoint(model=new_model, compression_state=TFCompressionState(new_compression_ctrl))
+    load_checkpoint(new_ckpt, ckpt_path)
+
+    builder = QuantizationBuilder(NNCFConfig())
+    builder.load_state(new_compression_state['builder_state'])
+    # pylint:disable=protected-access
+    new_quantizer_setup = builder._quantizer_setup
+
+    assert _quantization_setup_cmp(quantizer_setup, new_quantizer_setup)
+    assert new_compression_ctrl.get_state() == compression_ctrl.get_state()
+    assert tf.reduce_all([tf.reduce_all(w_new == w) for w_new, w in zip(new_model.weights, model.weights)])
+
+
+DUMMY_STR = 'dummy_str'
+
+
+def _quantization_setup_cmp(qs1: QuantizationSetup, qs2: QuantizationSetup):
+    if qs1.__class__ is qs2.__class__:
+        return all(_quantization_point_cmp(qp1, qp2) for qp1, qp2 in zip(iter(qs1), iter(qs2)))
+    return False
+
+
+def _quantization_point_cmp(qp1: QuantizationPoint, qp2: QuantizationPoint):
+    if qp1.__class__ is qp2.__class__:
+        return qp1.target_point == qp2.target_point and \
+               qp1.quantizer_spec == qp2.quantizer_spec and \
+               qp1.op_name == qp2.op_name
+    return False
+
+
+def _check_and_add_insertion_point(quantization_point_factory, setup, target_point):
+    point = quantization_point_factory(target_point=target_point)
+    setup.add_quantization_point(point)
+    check_serialization(target_point)
+    check_serialization(point, _quantization_point_cmp)
+
+
+GROUND_TRUTH_STATE = {
```

Comment:
backward compatibility test for the builder state.
should help to not run sota eval to find mistakes in builder state loading.